Feature/configure

jeffersonpuchalski opened the pull request:
- Create new configure.sh file.
- Add change to be able to merge.

Commit:
Adcionado logger e config file

jeffersonpuchalski committed:
Creating and refactoring configure file, Fix wrong changes in original code

jeffersonpuchalski committed:
Fix merge conflicts

wviana commented on configure.sh:
```diff
@@ -0,0 +1,35 @@
+}
+
+# Append all Information to Log file.
+function Logger () {
```
Couldn't log and config be on separated files ?

jeffersonpuchalski replied:
Yes, i will divide this.

wviana commented on configure.sh:
```diff
@@ -0,0 +1,35 @@
+# Append all Information to Log file.
+function Logger () {
+
+    echo "[$1] $DATE -> $2" &>> $log;
```
May be useful to write Logger stdin if there is one. Other wise write the second parameter. 
By this would be possible to send the curl stderr into a log with `curl ... 2| Logger info` something like this. 

PS: not so sure about 2| but sure that there is a way to pipe stderr.

jeffersonpuchalski replied:
yea, we can user Logger with any file descriptor(stdou, stdin, stderr). 
for working need redirect the desired file descriptor to function.
But we need establish some rules for log rotation, who i forgot :hankey:

wviana commented on pbcli.cfg:
```diff
@@ -0,0 +1 @@
+'log="pbcli.log"
```
Which other variables would you suggest to be placed in here? 
The only one that I thought about is base_url that is in http module.

jeffersonpuchalski replied:
Any fixed variable thought enviroment will no change, to be sourced.
Ex.:
base_url -> Fixed base url for project
log_directory -> Fixed log directory
conn_timeout -> Connection timeout.

wviana commented on .gitignore:
```diff
@@ -1 +1,2 @@
 *.swp
+*.log
```
Thanks for remembering.

jeffersonpuchalski replied:
U welcome :+1:

wviana reviewed (commented):
Not really problems.
Most talks about specific parts of code, so we may chat to better understanding.

wviana commented:
I'll wait for config and log file splitting before accepting the PR. 

Thank you so much for collaborating.

jeffersonpuchalski committed:
Create and refactoring logger.sh file.
Create configure file for loading configure file.
added logger function to main program.
added logger to projects function

jeffersonpuchalski committed:
Merge branch 'dev' into feature/configure
